src/main/java/core/basesyntax/service/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private static final String HANDLER_DOESNT_EXIST = "No handler found for operation: ";
+    private final Map<FruitTransaction.Operation, OperationStrategy> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationStrategy> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public void execute(FruitTransaction fruitTransaction) {
+        OperationStrategy handler = operationHandlers.get(fruitTransaction.getOperation());
+        if (handler != null) {
+            handler.execute(fruitTransaction);
+        } else {
+            System.out.println(HANDLER_DOESNT_EXIST + fruitTransaction.getOperation());
```

Comment:
Consider throwing a `RuntimeException` instead of printing a message when no handler is found for an operation. This will provide better error handling and make it easier to identify issues during execution.